Sources/AlgoliaSearch-Client/FilterBuilder/Builders/FilterBuilder.swift

```diff
@@ -0,0 +1,411 @@
+        let anyFilter = AnyFilter(filter)
+        guard let filtersForGroup = groups[group] else {
+            return false
+        }
+        return filtersForGroup.contains(anyFilter)
+    }
+    
+    func replace(_ attribute: Attribute, by replacement: Attribute, in group: AnyFilterGroup) {
+        guard let filtersForGroup = groups[group] else { return }
+        let filtersToReplace = filtersForGroup.filter { $0.attribute == attribute }
+        let filtersReplacements = filtersToReplace.map { $0.replacingAttribute(by: replacement) }
+        let updatedFilters = filtersForGroup.subtracting(filtersToReplace).union(filtersReplacements)
+        update(updatedFilters, for: group)
+    }
+    
+    func replace<T: Filter, D: Filter>(filter: T, by replacement: D, in group: AnyFilterGroup) {
+        guard let filtersForGroup = groups[group] else { return }
+        let filtersToReplace = filtersForGroup.filter { AnyFilter(filter) == $0 }
+        let filtersReplacement = [AnyFilter(replacement)]
+        let updatedFilters = filtersForGroup.subtracting(filtersToReplace).union(filtersReplacement)
+        update(updatedFilters, for: group)
+    }
+    
+    func replace<T: Filter>(_ filter: T, by replacement: T) {
+        groups.keys.forEach { group in
+            replace(filter: filter, by: replacement, in: group)
+        }
+    }
+    
+    func move<T: Filter>(filter: T, from origin: AnyFilterGroup, to destination: AnyFilterGroup) -> Bool {
+        if remove(filter, from: origin) {
+            add(filter, to: destination)
+            return true
+        }
+        return false
+    }
+    
+    @discardableResult func remove<T: Filter>(_ filter: T, from anyGroup: AnyFilterGroup) -> Bool {
+        return removeAll([filter], from: anyGroup)
+    }
+    
+    @discardableResult func removeAll<T: Filter, S: Sequence>(_ filters: S, from anyGroup: AnyFilterGroup) -> Bool where S.Element == T {
+        let filtersToRemove = filters.map(AnyFilter.init)
+        guard let existingFilters = groups[anyGroup], !existingFilters.isDisjoint(with: filtersToRemove) else {
+            return false
+        }
+        let updatedFilters = existingFilters.subtracting(filtersToRemove)
+        update(updatedFilters, for: anyGroup)
+        return true
+    }
+    
+    func removeAll(from group: AnyFilterGroup) {
+        groups.removeValue(forKey: group)
+    }
+    
+    func removeAll(for attribute: Attribute, from group: AnyFilterGroup) {
+        guard let filtersForGroup = groups[group] else { return }
+        let updatedFilters = filtersForGroup.filter { $0.attribute != attribute }
+        update(updatedFilters, for: group)
+    }
+    
+    func toggle<T: Filter>(_ filter: T, in group: AnyFilterGroup) {
+        if contains(filter, in: group) {
+            remove(filter, from: group)
+        } else {
+            add(filter, to: group)
+        }
+    }
+
+    func build(_ group: AnyFilterGroup, ignoringInversion: Bool) -> String {
```

Comment:
I forgot what we decided in the end: Do we want it to return `String?` in case there are no filters?

Reply:
Yes, it must be nil. Fixed it everywhere for consistency.